File: r/main4.R
dangerous <- function(v1, v2){ if(v2==0L) stop("divide by zero"); v1 %/% v2 }
foo <- function(array, counter){
  if(counter==0L) return(dangerous(array[[1L]], counter))
  foo(array, counter-1L)
}
array <- as.integer(rep(0L,1000))
cat("The result is", foo(array, 6L), "\n")

File: javascript/main4.mjs
function dangerous(v1, v2){ if(v2===0) throw new Error("divide by zero"); return (v1/v2)|0; }
function foo(array, counter){ if(counter===0) return dangerous(array[0], counter); return foo(array, counter-1); }
const array = Array(1000).fill(0);
console.log(`The result is ${foo(array, 6)}`);

